### ISSUE TITLE
API calls will stop working some time in early 2025

### ISSUE THREAD

**Issue by Billiam** [Billiam/hardcoverapp.koplugin#6]
Hardcover has announced that **all API keys** will be reset some time in January of 2025.

When this occurs, API calls made with this plugin _will fail_, resulting in empty search results, reading status not updating and so on.

If you have installed this plugin before that date (ETA unknown), you _will need to_ update your API key, following the [installation instructions in the readme](https://github.com/Billiam/hardcoverapp.koplugin?tab=readme-ov-file#installation)

You must update your API key _after this reset_. It cannot be updated early

This issue will be updated again after this reset has occurred to include relevant dates and instructions (if any)

**Comment by Philantrop** [comment]
This change has occurred and necessitated updating config.lua/hardcover_config.lua with the updated token which can be found at https://hardcover.app/account/api while logged-in. 

I suggest closing this issue.

**Comment by Billiam** [comment]
This is actually a warning about a still to come change that has now been delayed by hardcover.

**Comment by Philantrop** [comment]
I know - because my API key was reset this month. :-)

**Comment by masonrhayes** [comment]
My API key was also reset about 3 days ago.

**Comment by Billiam** [comment]
Here's the most recent update about the reset that I'm aware of. As of yesterday, their rails migration was still in progress, but getting closer. I am still expecting a universal passkey reset, which is why this issue was created and pinned.

![Discord chat screenshot. Adam Fortuna (1/18/25, 10:23am): The reset will still happen, but not until the rails migration. We still have about 40 bugs to work through before that. Once it's about a week out from release, I'm planning to email everyone who's use d the API in the last month (now that we have the proxy, we know this), and let them know.](https://github.com/user-attachments/assets/400fbe5e-5b1c-4e46-9e96-98e7aaf1181c)

**Comment by Billiam** [comment]
This has been postponed until Hardcover is open sourced. Will create a new issue when/if that looks imminent.
